### PR TITLE
Update markdown to 3.1.1

### PR DESCRIPTION
**Pull request by philip-khor**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by philip-khor** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: {{ sha256 }}
+  sha256: "2e50876bcdd74517e7b71f3e7a76102050edec255b3983403f1a63e7c8a41e7a"
 
 build:
   noarch: python
+  skip: true
```
Comment (dbast) [inline recipe/meta.yaml]: Remove that skip and rerender afterwards. Otherwise all builds are skipped.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -23,11 +23,14 @@ requirements:
...
   run:
     - python
+    - setuptools >=36
 
 test:
   imports:
     - markdown
     - markdown.extensions
+    - coverage
+    - pyyaml
```
Comment (dbast) [inline recipe/meta.yaml]: Why coverage and pyyaml here?

Comment (philip-khor) [reply recipe/meta.yaml]: these packages are listed under testing in extras_require in setup.py. Should I omit these instead?

Comment (dbast) [reply recipe/meta.yaml]: Well, those would be extra requirements if you would run the unittests here ... but `imports` have nothing to do with running unittests. Thus they can be removed here.

Comment (dbast) [reply recipe/meta.yaml]: See also:
* https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#test-requirements 
* https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#test-commands 

But that is not required here.

Comment (dbast) [reply recipe/meta.yaml]: The import tests are just doing something like `python -c "import markdown, markdown.extensions"`... which is normally enough.

**Comment by philip-khor** [comment]
@conda-forge-admin, please rerender

**Pull request merged**